### DISCUSSION TITLE
PurePosixPath subclass causing Mismatch between signature..  since 1.1.373

### DISCUSSION THREAD

**Discussion by davetapley** [microsoft/pyright#8525] [Q&A] [answered]
`1.1.354` was happy with code below (and works at run time), but I just bumped to `1.1.373` and I'm getting:

```
error: Mismatch between signature of __new__ and __init__ in class "Sensor"
    Signature of __init__ is "(*parts: str | PurePosixPath) -> None"
    Signature of __new__ is "(*args: StrPath, **kwargs: Unused) -> Sensor" (reportInconsistentConstructor)
```

Offending code:

```py
from pathlib import PurePosixPath
from pathvalidate import Platform, sanitize_filepath

class Sensor(PurePosixPath):
    def __init__(self, *parts: str | PurePosixPath) -> None:
        parts_ = tuple(sanitize_filepath(str(p), platform=Platform.POSIX)
                       for p in parts)
        super(Sensor, self).__init__(*parts_)
```

I know there's been awkwardness around `pathlib` subclassing in the past, but I'm on Python 3.12.1 and I figured it had all been cleaned up. Any ideas? 🤔

**Comment by erictraut** [comment, marked as answer]
I presume that you have manually enabled the `reportInconsistentConstructor` check or enabled "strict" type checking mode in your configuration. This diagnostic check is not enabled by default. If you don't want errors of this type to be reported, you can turn it back off.

The purpose of this check is to report situations where the `__new__` method and `__init__` method for a class are inconsistent. That is the case in your example.

According to the typeshed stubs, the class `PurePosixPath` inherits a `__new__` method from `PurePath`, and it is defined as:

```python
    if sys.version_info >= (3, 12):
        def __new__(cls, *args: StrPath, **kwargs: Unused) -> Self: ...
    else:
        def __new__(cls, *args: StrPath) -> Self: ...
```

Since you're using Python 3.12, pyright will use the first of these two `__new__` definitions. 

If you want your `__init__` method to be compatible, it would also need to accept positional parameters of type `str | PathLike[str]` and a `**kwargs` parameter.
